client/pages/spotify-login.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+export default class SpotifyLogin extends React.Component {
+  constructor(props) {
+    super(props);
+    this.makeid = this.makeid.bind(this);
+    this.getRandomInt = this.getRandomInt.bind(this);
+    this.initiateSpotifyLogin = this.initiateSpotifyLogin.bind(this);
+    this.sha256 = this.sha256.bind(this);
+    this.base64urlencode = this.base64urlencode.bind(this);
```

Comment:
Double check which of these `.bind()`s is actually necessary. If none, then you can delete your constructor.